testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -33,7 +33,101 @@
 from MDAnalysis.analysis import base
 
 from MDAnalysisTests.datafiles import PSF, DCD, TPR, XTC
-from MDAnalysisTests.util import no_deprecated_call
+from MDAnalysisTests.util import no_deprecated_call, get_running_dask_client
+from MDAnalysis.lib.util import is_installed
+
+class FrameAnalysis(base.AnalysisBase):
+    """Just grabs frame numbers of frames it goes over"""
+
+    available_backends = ['local', 'dask', 'dask.distributed', 'multiprocessing']
+
+    def __init__(self, reader, **kwargs):
+        super(FrameAnalysis, self).__init__(reader, **kwargs)
+        self.traj = reader
+    
+    def _prepare(self):
+        self.results.found_frames = []
+
+    def _single_frame(self):
+        self.results.found_frames.append(self._ts.frame)
+    
+    def _conclude(self):
+        self.found_frames = list(self.results.found_frames)
+
+class IncompleteAnalysis(base.AnalysisBase):
+    def __init__(self, reader, **kwargs):
+        super(IncompleteAnalysis, self).__init__(reader, **kwargs)
+
+
+class OldAPIAnalysis(base.AnalysisBase):
+    """for version 0.15.0"""
+
+    def __init__(self, reader, **kwargs):
+        self._setup_frames(reader, **kwargs)
+
+    def _single_frame(self):
+        pass
+
+class FinallyRaised(Exception):
+    pass
+
+def square(x):
+    """Testing function for TestParallelExecutor 
+    added here to avoide pickle problems with multiprocessing
+
+    Parameters
+    ----------
+    x : int
+        number to be squared
+
+    Returns
+    -------
+    int
+        square of the number
+    """
+    return x**2
+
+class TestParallelExecutor:
+    @pytest.mark.parametrize(
+        "backend,n_workers,client",
+        [
+            (b, n, c)
+            for b in [
+                "local",
+                "multiprocessing",
+                "dask",
+                "dask.distributed",
+                "nonexistent_module_X",
+                None,
+            ]
+            for n in (-1, None, 1, 2, 3)
+            for c in (None, get_running_dask_client(), object)
+        ],
+    )
+    def test_combinations(self, backend, n_workers, client):
+        failing_conditions = [
+            backend not in base.ParallelExecutor.available_backends,
+            backend is None and client is None,
+            backend is not None and client is not None,
+            backend == 'dask.distributed',
+            n_workers is None,
+            isinstance(n_workers, int) and n_workers < 0,
+            not is_installed(backend, ignore_names=['local', None])
+        ]
+        if any(failing_conditions):
+            raised = ValueError
+        else:
+            raised = FinallyRaised
```

Comment:
```suggestion
        raised = ValueError if any(failing_condition) else FinallyRaised
```
? Clearer/more concise?

Reply:
indeed, I keep forgetting about this idiome.